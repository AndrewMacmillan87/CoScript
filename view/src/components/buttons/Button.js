import React from "react";
import PropTypes from "prop-types";

import "./buttons.css";

const Button = ({ onClick, id, classes, value }) => {
    return (
        <button onClick={onClick} id={id} className={classes} type='button'>
            <span className='btn-text'>
                <b>{value}</b>
            </span>
        </button>
    );
};

Button.propTypes = {
    onClick: PropTypes.func,
    classes: PropTypes.string,
    id: PropTypes.string,
    value: PropTypes.string
};

export default Button;
