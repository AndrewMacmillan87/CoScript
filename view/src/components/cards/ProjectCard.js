import React from "react";

import Button from "../buttons/Button";

import PropTypes from "prop-types";

import "./project-card.css";

const ProjectCard = ({
    src,
    img_alt,
    title,
    project_admin,
    genres,
    published,
    onClick
}) => {
    return (
        <div className='card-item'>
            <div className='card-box-image-container'>
                <img className='card-box-image' src={src} alt={img_alt} />
            </div>
            <div className='card-item-details'>
                <p>{title}</p>
                <p>{project_admin}</p>
                <p className='genres'>{genres}</p>

                <Button
                    classes='small-btn'
                    value={published ? "Read" : "Details"}
                    onClick={onClick}
                />
            </div>
        </div>
    );
};

ProjectCard.propTypes = {
    src: PropTypes.string,
    img_alt: PropTypes.string,
    title: PropTypes.string,
    project_admin: PropTypes.string,
    genres: PropTypes.string
};

export default ProjectCard;
