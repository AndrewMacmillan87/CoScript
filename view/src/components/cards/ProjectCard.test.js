import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import ProjectCard from "./ProjectCard";
import Button from "../buttons/Button";

describe("ProjectCard component", () => {
    test("renders without crashing", () => {
        global.scrollTo = jest.fn();
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <ProjectCard />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        global.scrollTo = jest.fn();
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <ProjectCard />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a div element with a className of card-item", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <ProjectCard />
                </Router>
            </Provider>
        );
        expect(wrapper.find({ className: "card-item" }).length).toEqual(1);
    });

    it("should render a div element with a className of card-box-image-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <ProjectCard />
                </Router>
            </Provider>
        );
        expect(
            wrapper.find({ className: "card-box-image-container" }).length
        ).toEqual(1);
    });

    it("should render a div element with a className of card-item-details", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <ProjectCard />
                </Router>
            </Provider>
        );
        expect(wrapper.find({ className: "card-item-details" }).length).toEqual(
            1
        );
    });

    it("should render 3 p elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <ProjectCard />
                </Router>
            </Provider>
        );
        expect(wrapper.find("p").length).toEqual(3);
    });

    it("should render a Button component on mount", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <ProjectCard />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Button).length).toEqual(1);
    });
});
