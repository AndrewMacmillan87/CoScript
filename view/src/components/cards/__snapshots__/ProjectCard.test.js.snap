// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard component renders correctly 1`] = `
<div
  className="card-item"
>
  <div
    className="card-box-image-container"
  >
    <img
      className="card-box-image"
    />
  </div>
  <div
    className="card-item-details"
  >
    <p />
    <p />
    <p
      className="genres"
    />
    <button
      className="small-btn"
      type="button"
    >
      <span
        className="btn-text"
      >
        <b>
          Details
        </b>
      </span>
    </button>
  </div>
</div>
`;
